### ISSUE TITLE
International Mirrors

### ISSUE THREAD

**Issue by skorfmann** [rubygems/rubygems.org#644]
I've been travelling across Australia and South-East Asia in the past 18 months and installing Gems was quite a hassle sometimes. Currently, I'm located in Vietnam which has particularly bad international internet connections. I reckon that the major problem is the huge latency to the S3 storage backend in the US.

I think it would be a great improvement to provide the possibility of regional mirrors for the data and it would be another step to a mature Ruby world.

I'm certainly [not the only one](http://binarymentalist.com/post/1314642927/proposal-we-have-to-create-a-rubygems-mirror-network) who is keen to make this happen.

Have there been plans like this already?

**Comment by crhan** [comment]
is there any updates? @dwradcliffe

**Comment by arthurnn** [comment]
@dwradcliffe isnt this done already?

**Comment by simi** [comment]
@arthurnn It [looks like](https://github.com/rubygems/rubygems-aws/commits/master/chef/site-cookbooks/rubygems/templates/default/nginx_balancer.conf.erb) it is not.

**Comment by dwradcliffe** [comment]
Nope, sorry, nothing changed here yet. This is pretty much on hold until RailsConf.

**Comment by qrush** [comment]
What's left on this? Should this issue be in rubygems/rubygems-aws instead?

**Comment by dwradcliffe** [comment]
Our plan is to shut down the artisanally created mirrors and move more of our traffic behind Fastly's network. This will put more data at the edge locations which should be faster for everyone. This will provide much more value than our mirrors ever did with much greater reliability.

This also paves the way for the new index and dependency api, which is currently under development.

We have a couple hurdles to get though before we can do this, primarily rebuilding our stats system so we can continue to track download stats for gems.

**Comment by quentindemetz** [comment]
Over the past couple of days, I've been running into a number of 503 errors when trying to install gems to deploy my application - so much that I haven't been able to achieve a single `bundle install` run without errors.
The errors seem to be coming from Fastly.

Example error messages (from different runs):
**actionview**
```
Gem::RemoteFetcher::FetchError: bad response Service Unavailable 503 (http://rubygems.global.ssl.fastly.net/gems/actionview-4.1.4.gem)
```
**marginalia**
```
Gem::RemoteFetcher::FetchError: bad response Service Unavailable 503 (http://rubygems.global.ssl.fastly.net/gems/marginalia-1.1.5.gem)
```

**Comment by dwradcliffe** [comment]
@quentindemetz We're working with Fastly right now to figure out what is happening. Unfortunately I don't have a workaround for you right now. :(

**Comment by dwradcliffe** [comment]
@quentindemetz we rolled out a change yesterday that should have helped

**Comment by quentindemetz** [comment]
Tried once, it worked flawlessly! Obviously not statistically significant, but I'll let you know if I run into any issues going forward.

**Comment by YiniYin** [comment]
fastly seems to be a bit intermittent as I just tried it now and it's gone through where a few minutes ago threw a 503 error.

**Comment by dwradcliffe** [comment]
Fastly has been stable lately and we have much better worldwide coverage than in years past. We're rolling out more Fastly things soon which will make everything even faster. I'm going to close this since we're not pursuing any other mirrors at this time.

**Comment by crhan** [comment]
fastly is almost blocked in China and we have launched two mirror sites for gem download speed up:

1. https://gems.ruby-china.org/
2. https://ruby.taoboa.org/

**Comment by JuanitoFatas** [comment]
> https://ruby.taoboa.org/

:point_right: https://ruby.taobao.org/

**Comment by cicku** [comment]
I still wonder why can't rubygems.org setup rsync and provide access to those wanna setup a mirror like those setup for Linux distributions.

Provided that Fastly, S3, Google CDN, MaxCDN, KeyCDN, Cloudfront or whatever are robust enough, however for mainland Chinese, all these non-local CDNs have been blocked due to some political websites using these services to propagate anti-government contents. Fastly is just equate to **Slowly**.

Please consider rsync support.

**Comment by indirect** [comment]
@cicku this is a closed issue. If you are having problems and want our help to try to fix them, feel free to open an issue of your own.
